src/dbnode/encoding/types.go

```diff
@@ -27,20 +27,33 @@ import (
 	"github.com/m3db/m3/src/dbnode/ts"
 	"github.com/m3db/m3/src/dbnode/x/xio"
 	"github.com/m3db/m3/src/dbnode/x/xpool"
-	"github.com/m3db/m3/src/x/serialize"
 	"github.com/m3db/m3/src/x/checked"
 	"github.com/m3db/m3/src/x/ident"
 	"github.com/m3db/m3/src/x/pool"
+	"github.com/m3db/m3/src/x/serialize"
 	xtime "github.com/m3db/m3/src/x/time"
 )
 
+// StreamOpts is an options struct that can be passed to Encoder.Stream()
+// to modify its behavior.
+type StreamOpts struct {
```

Comment:
super nit: If you audit the other "options" structs we tend to name the vars "opts" but the actual struct itself as "FooOptions", so perhaps we can name this `StreamOptions`?

Reply:
Sure. I just did this because it helps keep things on one line since we have to declare a lot of empty versions of these but happy to change it